stl/inc/chrono

```diff
@@ -2426,7 +2557,17 @@ namespace chrono {
             _Unique_lock _Lk(_Tzdb_mutex);
             auto [_Leap_sec, _All_ls_positive] = _Xtzdb_generate_leap_seconds(_Tzdb_list.front().leap_seconds.size());
             if (!_Leap_sec.empty()) {
-                _Tzdb_list.emplace_front(tzdb{_STD move(_Leap_sec), _All_ls_positive});
+                const auto& _Tzdb = _Tzdb_list.front();
+                vector<time_zone> _Zones;
+                _STD transform(_Tzdb.zones.begin(), _Tzdb.zones.end(), _STD back_inserter(_Zones),
+                    [](const auto& _Tz) { return time_zone{_Tz.name()}; });
+                vector<time_zone_link> _Links;
+                _STD transform(
+                    _Tzdb.links.begin(), _Tzdb.links.end(), _STD back_inserter(_Links), [](const auto& _Link) {
+                        return time_zone_link{_Link.name(), _Link.target()};
+                    });
+                _Tzdb_list.emplace_front(
+                    tzdb{_Tzdb.version, _STD move(_Zones), _STD move(_Links), _STD move(_Leap_sec), _All_ls_positive});

```

Comment:
Since this is the case when we are actually updating information in the `tzdb` (because there are new `leap_seconds` to keep track of) we should change the version of the `tzdb` before emplacing it back (since things like `reload_tzdb()` and `remote_version()` use the version to differentiate between different information they contain).  Right now we're getting the version from ICU, but I don't think that'll be sufficient because we'll also need to factor in leap seconds data.  We need to come up with a better versioning system here...

I don't want this to hold up this PR because there is other work built on top of this, so I'm also fine with filing a card in the chrono project to revisit this in a follow-up PR.

Reply:
sounds good to me, lets revisit this